### PR TITLE
OSASINFRA-3360: Add upstream/downstream sync for CAPO

### PR DESCRIPTION
**Pull request by mandre**
This commit add periodic jobs to sync downstream branches of CAPO with their upstream counterparts.

### CODE REVIEW

File: ci-operator/jobs/shiftstack/merge-bot/shiftstack-merge-bot-main-periodics.yaml [hunk 1]
```diff
@@ -52,6 +52,59 @@ periodics:
...
+  decoration_config:
+    skip_cloning: true
+  extra_refs:
+  - base_ref: main
+    org: shiftstack
+    repo: merge-bot
+  labels:
+    ci.openshift.io/generator: prowgen
+    pj-rehearse.openshift.io/can-be-rehearsed: "true"
+  name: periodic-ci-shiftstack-merge-bot-main-cluster-api-provider-openstack
```
Comment (EmilienM) [inline ci-operator/jobs/shiftstack/merge-bot/shiftstack-merge-bot-main-periodics.yaml:68]: do we already have a slack notification by `merge-bot`? If not we need to add it here.

Comment (mandre) [reply ci-operator/jobs/shiftstack/merge-bot/shiftstack-merge-bot-main-periodics.yaml:68]: we do, for example the rehearsal sent a message on slack.

**Comment by mandre** [comment]
/hold
I've broken the stable branch sync with https://github.com/openshift/release/pull/47589/. Working on a fix...

**Comment by EmilienM** [comment]
/lgtm
@mandre whenever you think it's ready, hold cancel.
Thanks for this work!

**Comment by mandre** [comment]
Thanks for the review @EmilienM. It's ready, I simply forgot to remove the hold.
/hold cancel

**Pull request merged**